lib/api/core/models/security/profile.js

```diff
@@ -1,118 +1,143 @@
+      .then(roles => {
+        return new Bluebird(resolve => {
+          async.some(roles, (role, callback) => {
+            role.isActionAllowed(request, kuzzle)
+              .then(isAllowed => callback(null, isAllowed));
+          }, (error, result) => resolve(result));
+        });
+      });
   }
 
-  return Promise.resolve(true);
-};
+  /**
+   * Validates the Profile format
+   *
+   * @return {Promise<boolean>}
+   */
+  validateDefinition() {
+    if (!_.isArray(this.policies)) {
+      return Bluebird.reject(new BadRequestError('The roles member must be an array'));
+    }
+
+    if (this.policies.length === 0) {
+      return Bluebird.reject(new BadRequestError('The roles member array cannot be empty'));
+    }
+
+    return Bluebird.resolve(true);
+  }
 
-/**
- * Resolves an array of rights related to the profile's roles.
- *
- * @return {Promise}
- */
-Profile.prototype.getRights = function profileGetRights (kuzzle) {
-  var profileRights = {};
-
-  return this.getRoles(kuzzle)
-    .then(roles => {
-      roles.forEach(role => {
-        var restrictedTo = _.cloneDeep(role.restrictedTo);
-
-        if (restrictedTo === undefined || restrictedTo.length === 0) {
-          restrictedTo = [{index: '*', collections: ['*']}];
-        }
-
-        Object.keys(role.controllers).forEach(controller => {
-          Object.keys(role.controllers[controller].actions).forEach(action => {
-            var actionRights = role.controllers[controller].actions[action];
-
-            restrictedTo.forEach(restriction => {
-              if (restriction.collections === undefined || restriction.collections.length === 0) {
-                restriction.collections = ['*'];
-              }
-
-              restriction.collections.forEach(collection => {
-                var
-                  rightsObject = {},
-                  rightsItem = {
-                    controller: controller,
-                    action: action,
-                    index: restriction.index,
-                    collection: collection
-                  },
-                  rightsKey = highwayhash.asHexString(kuzzle.hashKey, Buffer.from(JSON.stringify(rightsItem)));
-
-                rightsItem.value = actionRights;
-                rightsObject[rightsKey] = rightsItem;
-                _.assignWith(profileRights, rightsObject, Policies.merge);
+  /**
+   * Resolves an array of rights related to the profile's roles.
+   *
+   * @return {Promise}
+   */
+  getRights(kuzzle) {
+    const profileRights = {};
+
+    return this.getRoles(kuzzle)
+      .then(roles => {
+        roles.forEach(role => {
+          let restrictedTo = _.cloneDeep(role.restrictedTo);
+
+          if (restrictedTo === undefined || restrictedTo.length === 0) {
+            restrictedTo = [{index: '*', collections: ['*']}];
+          }
+
+          Object.keys(role.controllers).forEach(controller => {
+            Object.keys(role.controllers[controller].actions).forEach(action => {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 81 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)